text/0010-exception-types.md

```diff
@@ -0,0 +1,119 @@
+* Start Date: 2022-09-17
+* RFC Type: feature
+* RFC PR: 0010
+* RFC Status: draft
+
+# Summary
+
+This RFC suggests a feature which introduces additional types of exceptions next to `mechanism.handled`.
+
+# Motivation
+
+Currently, exception which cause the running software to exit (the process died/hard crash) are marked as `handled: false`. This isn't enough for SDKs where an exception can be unhandled but at the same time doesn't cause the software to exit.
+
+Sentry shows exceptions which aren't caught by the developer (unhandled) but also do not cause the software to exit in the same way as exceptions which are manually caught by the developer. This seems rather unintuitive and makes exceptions seem less severe than they are.
+
+This issue impacts for example the [Dart/Flutter SDK](https://github.com/getsentry/sentry-dart/issues/456), the Unity SDK, the React Native SDK and possibly more.
+
+Another issue is, that excpetions which don't cause the software to exit but are unhandled, are not considered in the [`session health` metric](https://develop.sentry.dev/sdk/sessions/).
```

Comment:
What do you mean by 'they are not considered'?

My understanding is that we mark a session as crashed for all unhandled errors. For example, on Browser, we finish the session as crashed if there's an unhandled error, even though there was no crash.

Is this referring to Flutter where in the event of a Dart unhandled exception we don't close the session as crashed? But if there's a native unhandled error (which terminates the app) we do?

Reply:
> My understanding is that we mark a session as crashed for all unhandled errors. For example, on Browser, we finish the session as crashed if there's an unhandled error, even though there was no crash.

That probably wouldn't be the case for Flutter, if it had sessions, based on the current behavior. Even for Flutter for Web.

> Is this referring to Flutter where in the event of a Dart unhandled exception we don't close the session as crashed? But if there's a native unhandled error (which terminates the app) we do?

Yes, exactly.

Based on your questions, I'm assuming that `crashed`/`unhandled` mean different things across different SDKs. That's one more reason to align across SDKs.

Reply:
> Based on your questions, I'm assuming that crashed/unhandled mean different things across different SDKs. That's one more reason to align across SDKs.

Yes, this RFC tries to align this inconsistency as well @ueman .

Sessions are only marked as crashes when it cashes on the Native side.